fbw-a32nx/src/systems/instruments/src/EFB/Ground/Pages/Payload/A380_842/A380Payload.tsx

```diff
@@ -479,6 +488,22 @@ export const A380Payload: React.FC<A380Props> = ({
         gsxPayloadSyncEnabled,
     ]);
 
+
+  // If totalPax is 0, deactivate all sounds from the cabin.
+    useEffect(() => {
+        const cabinSoundStatus:number = (totalPax > 0 ? 1 : 0);
+
+        if (globalSettingsRedux.passengerAmbienceSetting) {
+            setPassengerAmbienceEnabled(cabinSoundStatus);
+        }
+        if (globalSettingsRedux.cabinAnnouncementsSetting) {
+            setAnnouncementsEnabled(cabinSoundStatus);
+        }
+        if (globalSettingsRedux.boardingMusicSetting) {
+            setBoardingMusicEnabled(cabinSoundStatus);
+        }
+    }, [totalPax]);
```

Comment:
Sorry that I haven't been active to see this PR earlier but there's a bit of a flaw in this current method/implementation which is that the react hook would only update if the Payload page in the EFB is open, if the user switches away from it this would not work.

Therefore it would be better to implement this in the backend driver (Which is written in rust in payload.rs) which runs on the main event loop where it would be always synced regardless of the state of the EFB.

Reply:
See the PayloadManager struct in ``src/fbw_commons/systems/src/payload/mod.rs``

Reply:
Hey. thanks for the review. I will look at the rust driver and give an update

Reply:
It should be possible to achieve the same result without needing to constantly manage state which should be cleaner and easier to maintain too

Reply:
Yes ok this is possible, but i thought we have to implement this feature in the rust backend, so managing the state in the efb application is trivial?

Reply:
I meant implementing this in the rust systems backend would make this more trivial